### PR TITLE
Swap execution flow order in enqueue

### PR DESCRIPTION
**Pull request by smellthemoon**
Signed-off-by: lixinguo <xinguo.li@zilliz.com>
#21848

### CODE REVIEW

**Review by congqixia — COMMENTED** [review]
what if `Enqueue` succeeded but `addPChanStats` failed? Will the task execute despite of that user get an error?

File: internal/proxy/task_scheduler.go [hunk 1]
```diff
@@ -227,11 +227,11 @@ type dmTaskQueue struct {
...
 	queue.statsLock.Lock()
 	defer queue.statsLock.Unlock()
-	err := queue.baseTaskQueue.Enqueue(t)
+	err := queue.addPChanStats(t)
 	if err != nil {
 		return err
 	}
-	err = queue.addPChanStats(t)
+	err = queue.baseTaskQueue.Enqueue(t)
 	if err != nil {
```
Comment (yah01) [inline internal/proxy/task_scheduler.go:235]: replace the following line with `return err`

Comment (yah01) [reply internal/proxy/task_scheduler.go:235]: or `return queue.baseTaskQueue.Enqueue(t)`

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**